lib/Mojolicious.pm

```diff
@@ -41,14 +42,29 @@ has plugins            => sub { Mojolicious::Plugins->new };
 has preload_namespaces => sub { [] };
 has renderer           => sub { Mojolicious::Renderer->new };
 has routes             => sub { Mojolicious::Routes->new };
+has secrets_file       => sub { $ENV{MOJO_SECRETS_FILE} || shift->home->rel_file('mojo.secret') };
 has secrets            => sub {
   my $self = shift;
+  my $file = $self->secrets_file;
 
-  # Warn developers about insecure default
-  $self->log->trace('Your secret passphrase needs to be changed (see FAQ for more)');
+  if (-f $file) {
+    my @secrets = grep { length $_ >= 22 }    # 22 random chars in base62 alphabet should be at
+                                              # best 128 bits of entropy
+      split /\n/, path($file)->slurp;
 
-  # Default to moniker
-  return [$self->moniker];
+    die "Your secrets_file $file does not contain any acceptable secret (of 22 chars or more)" unless @secrets;
+
+    return [@secrets];
+  }
+
+  # If no secret exists, generate one and attempt to write it back to
+  # secret_file, taking care that the file is only readable by the current user.
+  my $secret = urandom_token;
+  path($file)->touch->chmod(0600)->spew($secret);
+
+  $self->log->trace("Your secret passphrase has been set to strong random value and stored in $file");
+
+  return [$secret];
```

Comment:
While I think this is generally a good and necessary step, this will invalidate session cookies in currently insecure apps so that will need to at minimum be noted in the changelog (which is not the responsibility of this PR).

Reply:
This also does not support our rolling secret feature, which i believe increases security if used well.